LSPS5/README.md

```diff
@@ -0,0 +1,512 @@
+
+* MUST use a HTTPS-protocol URI to the notification delivery
+  service.
+* MUST encode the following data in any part of the URI
+  (hostname, path, or `GET` parameters):
+  * The LSP node ID.
+  * Some authorization to allow the given LSP to wake the
+    client, ideally a proof that the client has a channel
+    or other relationship with the LSP.
+  * Some way to identify the target device the client is
+    running on.
+* MAY encode any additional information in the URI.
+
+> **Rationale** By placing most data in the URI, the LSP
+> interface is simplified and the client developer has the
+> flexibility to add any additional information they want
+> the LSP to simply pass through from the client instance
+> to the notification delivery service.
+
+The LSP:
+
+* MUST persist the given webhook and associate it with the
+  client node ID before responding to this call.
+* MUST use the `webhook` specified in the latest
+  `lsps5.set_webhook` call.
+* MAY forget the `webhook` after 7 days (7 x 24 hours) from
+  the definite closure of the last channel between the LSP
+  and the client.
+  * MUST treat a possible channel open, such as via a
+    reserved SCID for a JIT channel, as a channel open, for
+    purposes of remembering the `webhook` for a client.
+
+The `lsps5.set_webhook` call results in the empty object
+`{}` on success, and has the following errors defined
+(error `code` number in parenthesis):
+
+* `unsupported_protocol` (1) - The `webhook` is not an
+  HTTPS-protocol URI.
+  * Future revisions of this LSPS MAY allow other protocols than
+    HTTPS.
+    However, an LSP that does not support that protocol is
+    allowed to return this error, and the client MUST either
+    set up with a protocol that the LSP supports, or not have
+    webhook notifications.
+    An LSP MUST support HTTPS at the minimum.
+
+### Webhook Call
+
+The LSP SHOULD contact the given webhook URI, if:
+
+* The client has registered one via `lsps5.set_webhook`
+* *and* the client currently does not have a BOLT8 tunnel
+  with the LSP (i.e. it is currently not connected to the
+  LSP), except if the client has just successfully registered
+  a webhook.
+* *and* one of the following conditions is true:
+  * The client has just successfully registered a webhook
+    (i.e. successful call of `lsps5.set_webhook`); the LSP
+    SHOULD do this even if the client is currently connected
+    to the LSP.
+  * The LSP learns of an incoming payment to the client
+    (whether an incoming HTLC, or other future mechanism
+    for incoming payments).
+    * For HTLCs (or in the future, PTLCs), the LSP SHOULD hold
+      onto the incoming H/PTLC for a reasonable time that
+      allows the client to awaken, but short enough that the
+      rest of the published network is not unduly impacted.
+  * An HTLC or other time-bound contract, in either direction,
+    in a channel with the client, is about to time out.
+  * The LSP wants to take back some of its liquidity towards
```

Comment:
Yes, but it could also be e.g. just mutually closing a channel instead of uniltaterally closing it while the clinet is offline. If an LSP has already fulfilled its published obligation to keep the channel open, and wants to reuse the liquidity with a client that has been underutilizing it, then the LSP might want to get it back by closing the channel. Or the LSP has multiple channels and wants  to close one of them. That sort of thing, "take back liquidity" basically includes "close channel" and "splice out".

Reply:
Is current text fine or...?